src/core/annotation.js

```diff
@@ -2871,6 +2926,126 @@ class FreeTextAnnotation extends MarkupAnnotation {
 
     this.data.annotationType = AnnotationType.FREETEXT;
   }
+
+  static async createNewAnnotation(
+    xref,
+    evaluator,
+    task,
+    annotation,
+    baseFontRef,
+    results,
+    dependencies
+  ) {
+    const fontSize = annotation.fontSize;
+    const freetextRef = xref.getNewRef();
+    const freetext = new Dict(xref);
+    freetext.set("Type", Name.get("Annot"));
+    freetext.set("Subtype", Name.get("FreeText"));
+    freetext.set("CreationDate", `D:${getModificationDate()}`);
+    freetext.set("Rect", annotation.rect);
+    const da = `/Helv ${fontSize} Tf ${getPdfColor(annotation.color)}`;
+    freetext.set("DA", da);
+    freetext.set("Contents", annotation.value);
+    freetext.set("F", 4);
+    freetext.set("Border", [0, 0, 0]);
+    freetext.set("Rotate", 0);
+
+    if (annotation.user) {
+      freetext.set("T", stringToUTF8String(annotation.user));
+    }
+
+    const resources = new Dict(xref);
+    const font = new Dict(xref);
+    font.set("Helv", baseFontRef);
+    resources.set("Font", font);
+
+    const helv = await WidgetAnnotation._getFontData(
+      evaluator,
+      task,
+      {
+        fontName: "Helvetica",
+        fontSize,
+      },
+      resources
+    );
+
+    const [x1, y1, x2, y2] = annotation.rect;
+    const w = x2 - x1;
+    const h = y2 - y1;
+
+    const lines = annotation.value.split("\n");
+    const scale = fontSize / 1000;
+    let totalWidth = -Infinity;
+    const encodedLines = [];
+    for (let line of lines) {
+      line = helv.encodeString(line).join("");
+      encodedLines.push(line);
+      let lineWidth = 0;
+      const glyphs = helv.charsToGlyphs(line);
+      for (const glyph of glyphs) {
+        lineWidth += glyph.width * scale;
+      }
+      totalWidth = Math.max(totalWidth, lineWidth);
+    }
+
+    let hscale = 1;
+    totalWidth += 2;
+    if (totalWidth > w) {
+      hscale = w / totalWidth;
+    }
+    let vscale = 1;
+    const totalHeight = 1.2 * fontSize * lines.length;
+    if (totalHeight > h) {
+      vscale = h / totalHeight;
+    }
+    const fscale = Math.min(hscale, vscale);
+    const newFontSize = fontSize * fscale;
+    const buffer = [
+      "q",
+      `0 0 ${w.toFixed(2)} ${h.toFixed(2)} re W n`,
+      `BT`,
+      `1 0 0 1 2 ${h.toFixed(2)} Tm 0 Tc ${getPdfColor(annotation.color)}`,
+      `/Helv ${newFontSize} Tf`,
+    ];
+
+    const vShift = (-1.2 * newFontSize).toFixed(2);
```

Comment:
Can we perhaps use a constant, rather than a magical value?
In any case, some kind of comment explaining how the value was decided upon would be helpful here.